test/basic.js

```diff
@@ -70,6 +71,68 @@ test('closing a drive should not close the store in use', async function (t) {
   await ss.add(seeds.core.key, seeds.core.value)
 })
 
+test('rejected peer by firewall', async function (t) {
+  t.plan(1)
+
+  const { testnet } = await createResources(t)
+
+  const store = new Corestore(RAM)
+
+  const swarm = new Hyperswarm({ bootstrap: testnet.bootstrap })
+  t.teardown(() => swarm.destroy())
+
+  const ss = new SimpleSeeder(store, swarm)
+  t.teardown(() => ss.destroy())
+
+  const notAllowed = new Hyperswarm({ bootstrap: testnet.bootstrap })
+  t.teardown(async () => await notAllowed.destroy())
+
+  const core = store.get({ name: 'seedbee-firewall-test' })
+  const seedBee = new SeedBee(core)
+  await seedBee.ready()
+  seedBee.putProperty(SeedBee.ALLOWED_PEERS_METADATA_KEY, Buffer.alloc(32).toString('hex'))
+
+  notAllowed.join(seedBee.core.discoveryKey)
+  await notAllowed.flush()
+
+  ss.on('peer-rejected', (remotePublicKey) => {
```

Comment:
Would place the on-handler higher up, this way you don't have to reason about whether it could already trigger too early